### PR TITLE
Updating hypre to v2.18.2

### PR DESCRIPTION
**Pull request by jamiebramwell**
This updates HYPRE to v2.18.2.

### CODE REVIEW

File: .gitlab/ci/build_quartz.yml [hunk 1]
```diff
@@ -57,17 +57,6 @@ gcc_8_1_0 (PR build_on_quartz):
     HOST_CONFIG: "quartz-toss_3_x86_64_ib-gcc@8.1.0.cmake"
   extends: .pr_build_on_quartz
 
-# Intel C++ compiler has poor C++17 support and will no longer be
```
Comment (jamiebramwell) [inline .gitlab/ci/build_quartz.yml:60]: After talking with @white238 , we decided to delete this CI job. It muddles the test logs and will confuse users with an out-of-date and broken host config.

**Review by joshessman-llnl — APPROVED** [review]

**Comment by jamiebramwell** [comment]
LGTM

**Review by white238 — APPROVED** [review]

**Comment by jamiebramwell** [comment]
LGTM

**Comment by jamiebramwell** [comment]
LGTM

**Pull request merged**